v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/templates/bigtablechangestreamstopubsub/FailsafeModJsonToPubsubMessageTransformer.java

```diff
@@ -0,0 +1,311 @@
+      }
+
+      @Setup
+      public void setUp() {
+        try {
+          final TopicName projectTopicName =
+              TopicName.of(
+                  pubSubUtils.getDestination().getPubSubProject(),
+                  pubSubUtils.getDestination().getPubSubTopicName());
+          publisher = Publisher.newBuilder(projectTopicName).build();
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+      @Teardown
+      public void tearDown() {
+        try {
+          if (publisher != null) {
+            publisher.shutdown();
+            publisher.awaitTermination(5, TimeUnit.MINUTES);
+          }
+        } catch (Exception e) {
+          throw new RuntimeException(e);
+        }
+      }
+
+      public static ChangelogEntry mapJsonToChangelogEntry(String jsonString) throws Exception {
+        Schema schema = ChangelogEntry.getClassSchema();
+
+        // Create a Decoder for JSON data
+        Decoder decoder = DecoderFactory.get().jsonDecoder(schema, jsonString);
+
+        DatumReader<ChangelogEntry> reader = new SpecificDatumReader<>(schema);
+
+        ChangelogEntry changelogEntry = reader.read(null, decoder);
+        return changelogEntry;
+      }
+
+      public static ChangelogEntryProto mapJsonToChangelogEntryProto(String jsonString)
+          throws Exception {
+        ChangelogEntryProto.Builder builder = ChangelogEntryProto.newBuilder();
+        JsonFormat.parser().ignoringUnknownFields().merge(jsonString, builder);
+
+        // Build the protobuf message
+        ChangelogEntryProto changelogEntryProto = builder.build();
+        return changelogEntryProto;
+      }
+
+      @ProcessElement
+      public void processElement(ProcessContext context) {
+        LOG.info("Reading a failsafeModJsonString");
+        FailsafeElement<String, String> failsafeModJsonString = context.element();
+
+        LOG.info("");
+
+        try {
+          com.google.pubsub.v1.PubsubMessage pubSubMessage =
+              publishModJsonStringToPubsubMessage(failsafeModJsonString.getPayload());
+
+          context.output(pubSubMessage);
+        } catch (Exception e) {
+          LOG.error("Writing to transform dead letter out", e);
+          context.output(
+              transformDeadLetterOut,
+              FailsafeElement.of(failsafeModJsonString)
+                  .setErrorMessage(e.getMessage())
+                  .setStacktrace(Throwables.getStackTraceAsString(e)));
+        }
+      }
+
+      /* Schema Details:  */
+      private PubsubMessage publishModJsonStringToPubsubMessage(String modJsonString)
+          throws Exception {
+        String messageFormat = pubSubUtils.getDestination().getTopicMessageFormat();
+        String messageEncoding = pubSubUtils.getDestination().getTopicMessageEncoding();
+        Publisher publisher = null;
+        ByteArrayOutputStream byteStream = new ByteArrayOutputStream();
+
+
+        block:
```

Comment:
This is my first time I have ever see this label feature being used. I think it's unnecessary if you exit switch using an exception